src/Value.php

```diff
@@ -138,4 +142,23 @@ public function asNonEmptyStrings(): array
 
         return array_values($this->value);
     }
+
+    public function asDateTimeImmutable(?string $format = null): DateTimeImmutable
+    {
+        $value = $this->asNonEmptyString();
+
+        try {
+            $date = $format !== null
+                ? DateTimeImmutable::createFromFormat($format, $value)
+                : new DateTimeImmutable($value);
+        } catch (Exception $e) {
+            throw new TypeError(sprintf('"%s" is not a valid date string.', $value), 412, $e);
+        }
+
+        if(!$date instanceof DateTimeImmutable) {
+            throw new TypeError(sprintf('Invalid values "%s" and format "%s"', $value, $format ?? 'no format'), 412);
```

Comment:
Can we drop this no format and just print two different messages?

Reply:
Splitted